src/dali/plugin/input/rest/kucoin_futures.py

```diff
@@ -0,0 +1,280 @@
+
+                if amount_value > 0:
+                    # Gain
+                    # Example json
+                    # {
+                    #   "time": 1644444400000,
+                    #   "type": "RealisedPNL",
+                    #   "amount": 66.66666666,
+                    #   "fee": 0.0,
+                    #   "accountEquity": 12345.67890123,
+                    #   "status": "Completed",
+                    #   "remark": "ETHUSDTM",
+                    #   "offset": 123456,
+                    #   "currency": "USDT"
+                    # }
+
+                    # To do: need to replace transaction type from INCOME to GAIN
+                    # GAIN currently not implemented in rp2
+                    in_transactions.append(
+                        InTransaction(
+                            plugin=self.plugin_name(),
+                            unique_id=unique_id,
+                            raw_data=json.dumps(item),
+                            timestamp=timestamp,
+                            asset=currency,
+                            exchange=self.exchange_name(),
+                            holder=self.account_holder,
+                            transaction_type=Keyword.INCOME.value,
+                            spot_price=Keyword.UNKNOWN.value,
+                            crypto_in=str(amount_value),
+                            crypto_fee=None,
+                            fiat_in_no_fee=None,
+                            fiat_in_with_fee=None,
+                            fiat_fee=None,
+                            notes="Futures gain"
+                        )
+                    )
+
+                else:
+                    # Loss
+                    # Example json
+                    # {
+                    #   "time": 1644444400000,
+                    #   "type": "RealisedPNL",
+                    #   "amount": -66.66666666,
+                    #   "fee": 0.0,
+                    #   "accountEquity": 12345.67890123,
+                    #   "status": "Completed",
+                    #   "remark": "ETHUSDTM",
+                    #   "offset": 123456,
+                    #   "currency": "USDT"
+                    # }
+
+                    # To do: need to replace transaction type from SELL to LOSS
+                    # LOSS currently not implemented in rp2
+                    out_transactions.append(
+                        OutTransaction(
+                            plugin=self.plugin_name(),
+                            unique_id=unique_id,
+                            raw_data=json.dumps(item),
+                            timestamp=timestamp,
+                            asset=currency,
+                            exchange=self.exchange_name(),
+                            holder=self.account_holder,
+                            transaction_type=Keyword.SELL.value,
+                            spot_price="0",
```

Comment:
Why is spot price 0? This would cause a 0 fiat loss: in other words it would be a taxable event with no impact.

Reply:
I could be wrong, but this is my thought process.

If we set spot price to 0, it would incur ((acquisition price - spot price) * quantity) loss and will be considered as taxable event. If this is not how it works, at least that is my intention and I would like to know how to code that as I am not sure how to do that.

Reply:
I don't have Kucoin so I'm just going by the example JSON above: does the `amount` field contain the fiat amount of the loss (in other words is -66 the fiat difference between the price at acquisition of the future contract and the price at expiration)?

Reply:
-66 would be the amount of USDT realized loss (it isn't fiat amount). So if you had 3000 USDT before, you would be now left with 2934 USDT after closing your future position. 

If the amount is positive (e.g 66 not -66), then it would be the amount of USDT realized gain. If the currency was XBT, then it would be BTC realized gain. -66 amount with XBT currency would be, realized loss of 66 Bitcoins (66 * $ 17000 in fiat values at current price)

Reply:
Oh, I see, thanks for the explanation. What would happen if the loss is -66 USDT at expiration but the user never had any USDT?

Reply:
before that happens, the position will be liquidated.

Reply:
Could we just represent the differential loss as a SELL transaction of 66 USDT with the actual spot price (or UNKNOWN if the exchange doesn't provide it)? This way it would be treated as a normal loss by the tax engine.

Reply:
I am kind of confused as to why the spot price needs to be set to market price.

ex) 

1 30 USDT bought with USD at $ 1.00 spot price
2. Bought a future contract.
3. Contract got liquidated. 30 USDT realized loss reported.

So, are you suggesting step 3 to be represented as a sell (OutTransaction) with spot price of $1.00 ? Wouldn't this generate a tax report of 0 capital loss? 

My assumption was to represent the above as the following.
step 1 -> Buy InTransaction with amount 30, currency USDT, spot price $1.00
step 3-> Sell OutTransaction with amount 30, currency USDT, spot price $0.00

Reply:
OK, I see what you mean: you want to ensure the 30 USDT are modeled as lost, not sold and so by setting the spot price to 0 you achieve that. Sorry it took me a while to get it: it looks good to me. @macanudo527, do you have any more thoughts on this? If not I think we can move on.